doc/static_route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,546 @@
+|create route for a prefix with NH IP address, interface and NH VRF| verify route is created|
+|create route for a prefix with interface and NH VRF| verify route is created|
+|create route for a prefix with NH IP addresses and distance|    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance | verify route is created|
+|create route for a prefix with interface and distance | verify route is created|
+|create route for a prefix with NH IP addresses, NH VRF and distane |    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance| verify route is created|
+|create route for a prefix with interface, NH VRF and distance | verify route is created|
+|create route for a prefix with blackhole and distance |verify route is created
+|CLI: show ip route static|   verify static routes are displayed with flag "S" (i.e Static)|
+|Delete route for a prefix with NH IP addresses| verify the route is deleted|
+|Delete route for a prefix with NH IP address and interface| verify route is deleted|
+|Delete route for a prefix with interface only| verify route is deleted|
+|Delete route for a prefix with NH IP addresses and NH VRF|    verify the route is deleted|
+|Delete route for a prefix with NH IP address, interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with blackhole| verify route is deleted|
+
+| Test cases for management static routes |Test Result |
+| :------ | :----- |
+|create route in default instance  for a prefix with NH IP address and interface as Management0 | verify route is created|
+|create route in default instance  for a prefix with interface as Management0 | verify route is created|
+|create route in mgmt vrf  for a prefix with NH IP address and interface as in-band interface | verify route is created|
+|create route in mgmt vrf  for a prefix with NH IP address | verify route is created|
+|create route in mgmt vrf  for a prefix with in-band interface | verify route is created|
+|create route in data VRF for a prefix with NH IP addresses and NH VRF as management vrf|    verify the route is created|
+|create route in data VRF for a prefix with NH interface and NH VRF as management vrf|    verify the route is created|
+|create route in data VRF for a prefix with NH interface as Management0 and NH VRF as default|    verify the route is created|
+|create route in mgmt vrf for a prefix with blackhole and distance |verify route is created
+|CLI: show ip route vrf mgmt static|   verify static routes are displayed with flag "S" (i.e Static)|
+|Delete all the above configured routes| verify all route are deleted
+|CLI: show ip route vrf mgmt static|   verify no static routes that are deleted are displayed|
+
+| Negative test cases |Test Result |
+| :------ | :----- |
+|Try creating a route with mismatched prefix(IPV4) and address(IPV6)| test will fail|

```

Comment:
please elaborate this. Note that IPv4 prefiux with IPv6 next-hop is a valid route combination

Reply:
ipv4 and  ipv6 mix  match is not supported:-

sonic(config)# ip route 3.4.5.6/32
  A.B.C.D    IP gateway address
  INTERFACE  IP gateway interface name
     Vrf-RED Vlan100 Vrf-common Bridge Ethernet0 Ethernet4 Ethernet8
     Ethernet12 Ethernet16 Ethernet20 Ethernet24 Ethernet28 Ethernet32
     Ethernet36 Ethernet40 Ethernet44 Ethernet48 Ethernet52 Ethernet56
     Ethernet60 Ethernet64 Ethernet68 Ethernet72 Ethernet76 Ethernet80
     Ethernet84 Ethernet88 Ethernet92 Ethernet96 Ethernet100 Ethernet104
     Ethernet108 Ethernet112 Ethernet116 Ethernet120 Ethernet124 bcm0 bond0
     docker0 dummy dummy0 lo Loopback99 eth0 lo-m mgmt
  Null0      Null interface
  blackhole  Silently discard pkts when matched
  reject     Emit an ICMP unreachable when matched
sonic(config)# ip route 3.4.5.6/32 3ffe:506::/32
% Interface name 3ffe:506::/32 is not valid